### ISSUE TITLE
Adversarial Training Methods for Semi-Supervised Text Classification

### ISSUE THREAD

**Issue by msrks** [furukawa-ai/deeplearning_papers#19]
ICLR17: https://arxiv.org/abs/1605.07725

Adversarial training provides a means of regularizing supervised learning algorithms while virtual adversarial training is able to extend supervised learning algorithms to the semi-supervised setting. However, both methods require making small perturbations to numerous entries of the input vector, which is inappropriate for sparse high-dimensional inputs such as one-hot word representations. We extend adversarial and virtual adversarial training to the text domain by applying perturbations to the word embeddings in a recurrent neural network rather than to the original input itself. The proposed method achieves state of the art results on multiple benchmark semi-supervised and purely supervised tasks. We provide visualizations and analysis showing that the learned word embeddings have improved in quality and that while training, the model is less prone to overfitting.

**Comment by msrks** [comment]
TL;DR: 
text classificationにVATを適用するために、input vectorではなく、embedded vectorに対してVATのperturbationを加えてやる。SSLでSOTA。

<img width="840" alt="2017-05-07 23 12 52" src="https://cloud.githubusercontent.com/assets/9496670/25781924/49491994-337b-11e7-8971-dbd0d259bfd8.png">

**Comment by msrks** [comment]
<img width="637" alt="2017-05-07 23 13 21" src="https://cloud.githubusercontent.com/assets/9496670/25781943/8f8d2170-337b-11e7-841b-0d42055ca1db.png">

**Comment by msrks** [comment]
<img width="548" alt="2017-05-07 23 15 54" src="https://cloud.githubusercontent.com/assets/9496670/25781944/984152be-337b-11e7-85a8-aa091ec61b5b.png">
